docs(pharos-site): add toggle button group documentation (#191)
ithaka/pharos: packages/pharos-site/static/guidelines/toggle-button-group.docs.mdx

import PageSection from '../../src/components/statics/PageSection.tsx';
import BestPractices from '../../src/components/statics/BestPractices.tsx';

<PageSection
  isHeader
  title="Toggle Button Group"
  storyBookType="components"

>   description="Toggle button groups present related actions together to allow the user to select a mutually exclusive option."

Reviewer: ```suggestion
  description="Toggle button groups present related actions together to allow the user to select from mutually exclusive options."
```

></PageSection>

<PageSection title="Usage">
  <p>

>     Within a toggle button group, only one option can be selected at the time. Selecting one option

Reviewer: ```suggestion
    Within a toggle button group, only one option can be selected at a time. Selecting one option
```

    deselects all others.
  </p>
  <p>
    Use a toggle button group as a switch between different views. For example, use the component to
    allow users to switch between a list view and gallery view.
  </p>
</PageSection>

<PageSection title="Best practices">
  <BestPractices
    Do={
      <ul>
        <li>

>           Items within a toggle button group should be related and have a logical connection to each
>           other

Reviewer: This item lacks parallel structure with the others, which finish the sentence "Do ________". Maybe this would work? I think we don't have solid parallel structure for our dos/don'ts, so maybe I'm overthinking at this particular juncture.

```suggestion
          Group items within a toggle button group that are related and have a logical connection to each
          other
```

        </li>
        <li>
          Think about whether your toggle button group needs text and an icon, text only, or an icon
          only
        </li>
        <li>Use short and concise text for the labels (if applicable)</li>
        <li>
          Place the toggle button group in close proximity (ideally just above) to the area it
          controls
        </li>
      </ul>
    }
    Dont={
      <ul>
        <li>Don't group items that aren't clearly related</li>
        <li>Don't use an icon-only toggle if the actions aren't clearly understood</li>

>         <li>Don't use the button group component if there are more than three options</li>

Reviewer: ```suggestion
        <li>Don't use a toggle button group if there are more than three options</li>
```

        Gallery
      </PharosToggleButton>
      <PharosToggleButton id="list-mode-button" icon-left="view-list">
        List
      </PharosToggleButton>
    </PharosToggleButtonGroup>
  </PageSection>
  <PageSection title="Text only" subSectionLevel={1}>
    <PharosToggleButtonGroup>
      <PharosToggleButton id="gallery-mode-button">Gallery</PharosToggleButton>
      <PharosToggleButton id="list-mode-button">List</PharosToggleButton>
    </PharosToggleButtonGroup>
  </PageSection>
  <PageSection title="Icon only" subSectionLevel={1}>
    <PharosToggleButtonGroup>
      <PharosToggleButton id="gallery-mode-button" icon-left="view-gallery" />
      <PharosToggleButton id="list-mode-button" icon-left="view-list" />
    </PharosToggleButtonGroup>
  </PageSection>
</PageSection>

<PageSection title="Accessibility">
  <PageSection subSectionLevel={1} title="Relevant WCAG guidelines">
    <ul>
      <li>
        <PharosLink href="https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html">
          1.3.1 Info and Relationships A
        </PharosLink>
      </li>
    </ul>
  </PageSection>
  <PageSection title="Code expectations" subSectionLevel={1}>
    <ul>
      <li>

>         There is a <code>role="group"</code> tag on the parent element <code>&lt;div&gt;</code>

Reviewer: Also: **must** the parent be a div?

```suggestion
        There is a <code>role="group"</code> attribute on the parent element <code>&lt;div&gt;</code>
```

Author: Good point. I just followed the doc but we can change it

Reviewer: We could probably just shorten this to "...on the parent element" in that case!

        <ul>
          <li>
            If the toggle button group is part of a toolbar use <code>role="toolbar"</code>
          </li>
        </ul>
      </li>
      <li>
        Use the semantic <code> &lt;button&gt; </code> on all grouped elements under the parent
        element
      </li>
      <li>
        If the button needs additional information in order to be understood by assistive

>         technologies, use an aria-label to convey this information

Reviewer: Either "use an `aria-label` attribute" or "use an ARIA label" might be better here.
